### ISSUE TITLE
Feature request: support accessing the original asset info in hooks

### ISSUE THREAD

**Issue by madflow** [webdeveric/webpack-assets-manifest#101]
## The problem

* As far as I can see - it is not possible to access the `info` of the original asset when using hooks.
* See: https://webpack.js.org/api/stats/#asset-objects
* Information in this object can however be used to create more predictive asset keys - when using the `copy-webpack-plugin`. Maybe also other plugins adding assets.

Feature request: Include the `info` object of the original asset and make it accessible in the applicable hooks.

Alternatively: Make the original asset object accessible.

## Details

* https://github.com/webpack-contrib/copy-webpack-plugin/issues/538
* https://github.com/webpack-contrib/copy-webpack-plugin/pull/542/files

**Comment by webdeveric** [comment]
@madflow The ability to see the current asset has existed in the `customize()` hook for a while now.
In the current version of this plugin (Webpack 4 compatible), it is the actual source object so it wasn't used very frequently.

The asset info object was added in Webpack 5 so I've updated `customize()` so that `asset` is an entry from `compilation.getAssets()`. 

You will be able to do something like this once I release an updated version.

```js
customize(entry, original, manifest, asset) {
  console.log(asset.info);
}
```

I've also updated the plugin to account for `info.sourceFilename` (`compiler.context` relative path).

You shouldn't need to do anything to have assets show up in the assets manifest when they were added with `copy-webpack-plugin` (or any other plugin that provides `info.sourceFilename`).
